core/src/main/java/com/linecorp/armeria/common/util/ShutdownHooks.java

```diff
@@ -87,14 +94,16 @@ private static CompletableFuture<Void> addClosingTask(
                 closeFuture.completeExceptionally(cause);
             }
         };
-        synchronized (autoCloseableOnShutdownTasks) {
+
+        reentrantLock.lock();
+        try {
             final Queue<Runnable> onShutdownTasks =
                     autoCloseableOnShutdownTasks.computeIfAbsent(autoCloseable, key -> new ArrayDeque<>());
             onShutdownTasks.add(task);
             if (!addedShutdownHook) {
                 Runtime.getRuntime().addShutdownHook(THREAD_FACTORY.newThread(() -> {
                     autoCloseableOnShutdownTasks.forEach((factory, queue) -> {
```

Comment:
Let's also add a lock here.

Reply:
I modified it according to the review, then the test time increased a lot
Did i fix something wrong?
```java
        reentrantLock.lock();
        try {
            final Queue<Runnable> onShutdownTasks =
                    autoCloseableOnShutdownTasks.computeIfAbsent(autoCloseable, key -> new ArrayDeque<>());
            onShutdownTasks.add(task);
            if (!addedShutdownHook) {
                Runtime.getRuntime().addShutdownHook(THREAD_FACTORY.newThread(() -> {
                    reentrantLock.lock();
                    try {
                        autoCloseableOnShutdownTasks.forEach((factory, queue) -> {
                            for (; ; ) {
                                final Runnable onShutdown = queue.poll();
                                if (onShutdown == null) {
                                    break;
                                } else {
                                    onShutdown.run();
                                }
                            }
                        });
                    } finally {
                        reentrantLock.unlock();
                    }
                }));
                addedShutdownHook = true;
            }
        } finally {
            reentrantLock.unlock();
        }
```

Reply:
> then the test time increased a lot
Did i fix something wrong?

It looks fine to me. How much time was it increased after the change?

Reply:
It took more than 20 minutes...(It took too long, so I stopped😭)

Reply:
I added it because if I don't add the lock, the build will fail
p.s. hope it's just that my computer is slow🙏